### PR TITLE
Shultsedu 255 add dic

### PR DESCRIPTION
**Pull request by nadiiazhurba**

### CODE REVIEW

File: SideMenuFlyoutShellTestProject/ViewModels/BasePrinciplePageViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+using SideMenuFlyoutShellTestProject.Helpers;
+using SideMenuFlyoutShellTestProject.Interfaces;
+using SideMenuFlyoutShellTestProject.Models;
+using Xamarin.Forms;
+
+namespace SideMenuFlyoutShellTestProject.ViewModels
+{
+    public class BasePrinciplePageViewModel : INotifyPropertyChanged, IViewModel<Principle>
+    {
+        private Principle _model;
```
Comment (YuliaLoyko) [inline SideMenuFlyoutShellTestProject/ViewModels/BasePrinciplePageViewModel.cs:15]: we also use a little bit another file structure:
all fields 
consttructor
all properties
all methods

and inside the groups: 
public
proteced
private

File: SideMenuFlyoutShellTestProject/AppShell.xaml.cs [hunk 2]
```diff
@@ -1,29 +1,59 @@
...
+            var iPrinciplePage = App.Container.GetInstance<PrincipleInfoPage>();
+            var dPrinciplePage = App.Container.GetInstance<PrincipleInfoPage>();
+
+            sPrinciplePage.ViewModel.Model = PrinciplesData.GetSPrincipleModel();
+            oPrinciplePage.ViewModel.Model = PrinciplesData.GetOPrincipleModel();
+            lPrinciplePage.ViewModel.Model = PrinciplesData.GetLPrincipleModel();
+            iPrinciplePage.ViewModel.Model = PrinciplesData.GetIPrincipleModel();
+            dPrinciplePage.ViewModel.Model = PrinciplesData.GetDPrincipleModel();
+
+            sPrinciplePage.Title = sPrinciplePage.ViewModel.Model.Title;
```
Comment (YuliaLoyko) [inline SideMenuFlyoutShellTestProject/AppShell.xaml.cs:46]: may be it's worth just binding the title?

File: SideMenuFlyoutShellTestProject/App.xaml.cs [hunk 3]
```diff
@@ -1,15 +1,43 @@
...
+            Container = new Container();
+            ConfigureContainer();
+        }
+
+        private static void ConfigureContainer()
+        {
+            var asm = Assembly.GetExecutingAssembly();
+
+            Container.Collection.Register(typeof(IView<>), asm);
+            Container.Collection.Register(typeof(IViewModel<>), asm);
```
Comment (YuliaLoyko) [inline SideMenuFlyoutShellTestProject/App.xaml.cs:30]: i'm a little bit confused with registering both interfaces and concrete viewModels. why would we need both?

**Review by YuliaLoyko — CHANGES_REQUESTED** [review]

File: SideMenuFlyoutShellTestProject/ViewModels/BasePrinciplePageViewModel.cs (hunk 1 above)
Comment (nadiiazhurba) [reply SideMenuFlyoutShellTestProject/ViewModels/BasePrinciplePageViewModel.cs:15]: Re-done order in all of the files.

File: SideMenuFlyoutShellTestProject/Views/PrincipleInfoPage.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿using SideMenuFlyoutShellTestProject.Interfaces;
+using SideMenuFlyoutShellTestProject.Models;
+using SideMenuFlyoutShellTestProject.ViewModels;
+
+namespace SideMenuFlyoutShellTestProject.Views
+{
+    public partial class PrincipleInfoPage : IView<Principle>
```
[the remark this answers is not in the document]
Comment (nadiiazhurba) [reply SideMenuFlyoutShellTestProject/Views/PrincipleInfoPage.xaml.cs:7]: Done it. Added base class for all pages.
And base ViewModel as well

File: SideMenuFlyoutShellTestProject/Views/PrincipleInfoPage.xaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                             VerticalOptions="FillAndExpand">
+                    <Image Source="{Binding Image}" 
+                           VerticalOptions="Center" 
+                           HeightRequest="200" />
+                </ContentView>
+            </StackLayout>
+        </StackLayout>
+        <ScrollView Grid.Row="1">
+            <StackLayout Orientation="Vertical" 
+                         Padding="30,0,30,24"
```
[the remark this answers is not in the document]
Comment (nadiiazhurba) [reply SideMenuFlyoutShellTestProject/Views/PrincipleInfoPage.xaml:31]: Done. Added spaces for all Margins and Paddings.

File: SideMenuFlyoutShellTestProject/Views/DetailsPage.xaml.cs [hunk 6]
```diff
@@ -1,26 +1,26 @@
...
 using Xamarin.Forms;
 
 namespace SideMenuFlyoutShellTestProject.Views
 {
     [QueryProperty("PrevTitle", "title")]
-    public partial class Details : ContentPage
+    public partial class DetailsPage
     {
-        private string prevTitle;
+        private string _prevTitle;
```
[the remark this answers is not in the document]
Comment (nadiiazhurba) [reply SideMenuFlyoutShellTestProject/Views/DetailsPage.xaml.cs:11]: Ok, I see. Renamed all local fields removing underscore.

File: SideMenuFlyoutShellTestProject/App.xaml.cs (hunk 3 above)
Comment (nadiiazhurba) [reply SideMenuFlyoutShellTestProject/App.xaml.cs:30]: That's for the possible future needs, but I'll just remove it for now, so it won't confuse.

File: SideMenuFlyoutShellTestProject/AppShell.xaml.cs (hunk 2 above)
Comment (nadiiazhurba) [reply SideMenuFlyoutShellTestProject/AppShell.xaml.cs]: Yes, I've already made these bindings, but somehow forgot to remove these lines. Removed them.

File: SideMenuFlyoutShellTestProject/ViewModels/BasePrinciplePageViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+        public ICommand GoToDetailsCommand => new Command(async () => await NavigateToDetailsPageAsync());
+
+        protected async Task NavigateToDetailsPageAsync()
+        {
+            await Shell.Current.GoToAsync(RouteBuilder.GoToDetailsPage(Model?.Title));
+        }
+
+        [NotifyPropertyChangedInvocator]
+        protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
```
Comment (YuliaLoyko) [inline SideMenuFlyoutShellTestProject/ViewModels/BasePrinciplePageViewModel.cs:81]: i'd move this one to a BaseViewModel (believe it'll be helpful for all viewModels)

**Review by YuliaLoyko — CHANGES_REQUESTED** [review]

File: SideMenuFlyoutShellTestProject/Interfaces/IViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,7 @@
+namespace SideMenuFlyoutShellTestProject.Interfaces
+{
+    public interface IViewModel<TModel>
+    {
+        TModel Model { get; set; }
+    }
+}
```
[the remark this answers is not in the document]
Comment (nadiiazhurba) [reply SideMenuFlyoutShellTestProject/Interfaces/IViewModel.cs:7]: Checked. Configured

File: SideMenuFlyoutShellTestProject/ViewModels/BasePrinciplePageViewModel.cs (hunk 7 above)
Comment (nadiiazhurba) [reply SideMenuFlyoutShellTestProject/ViewModels/BasePrinciplePageViewModel.cs:81]: Done it

**Review by YuliaLoyko — APPROVED** [review]

**Pull request merged**